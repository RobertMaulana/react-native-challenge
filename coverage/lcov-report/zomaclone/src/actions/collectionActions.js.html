<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for zomaclone/src/actions/collectionActions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">zomaclone/src/actions</a> collectionActions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">87.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">83.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>5/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/7</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { GET_COLLECTIONS } from './constants'
import axios from 'axios'
&nbsp;
export const getCollections = collections =&gt; ({
  type: GET_COLLECTIONS,
  payload: collections
})
&nbsp;
export const fetchCollections = () =&gt; (
  dispatch =&gt;
  // axios.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&amp;count=50',
  // {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  // .then(function (response) {
  //   dispatch(getCollections(response.data.collections))
  // })
  // .catch(function (err) {
  //   console.log(err.message);
  // })
  fetch('https://developers.zomato.com/api/v2.1/collections?city_id=1&amp;count=50',
  {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  .then(res =&gt; res.json())
  .then(data =&gt; dispatch(getCollections(data.collections)))
  .catch(<span class="fstat-no" title="function not covered" >fu</span>nction (err) {
<span class="cstat-no" title="statement not covered" >    console.log(err.message);</span>
  })
)
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 04 2017 15:09:03 GMT+0700 (WIB)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
