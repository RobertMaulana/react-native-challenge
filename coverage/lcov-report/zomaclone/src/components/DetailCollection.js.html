<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for zomaclone/src/components/DetailCollection.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">zomaclone/src/components</a> DetailCollection.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, {Component} from 'react'
import {
  Image
} from 'react-native'
import moment from 'moment'
import { Container, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body } from 'native-base';
&nbsp;
class DetailCollection extends Component {
  constructor(props){
    super(props)
    this.state = {
    }
  }
  static navigationOptions = {
    title: 'Detail',
  }
  render(){
    const { params } = this.props.navigation.state;
    const restaurant = params.collection;
&nbsp;
    return (
      &lt;Container&gt;
        &lt;Content&gt;
            &lt;Card style={{ flex: 0 }}&gt;
              &lt;CardItem&gt;
                &lt;Left&gt;
                  &lt;Thumbnail style={{width: 50, height:50}} source={{uri: 'https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png'}}/&gt;
                  &lt;Body&gt;
                      &lt;Text&gt;{restaurant.title}&lt;/Text&gt;
                      &lt;Text note&gt;{ moment().format("DD MMM YYYY") }&lt;/Text&gt;
                  &lt;/Body&gt;
                &lt;/Left&gt;
              &lt;/CardItem&gt;
              &lt;CardItem&gt;
                &lt;Body&gt;
                  &lt;Image style={{ width: 340, height: 200}} source={{uri: restaurant.image_url}}/&gt;
                  &lt;Text&gt;
                      {restaurant.description}
                  &lt;/Text&gt;
                  &lt;Button transparent textStyle={{color: '#87838B'}}&gt;
                      &lt;Icon name="logo-facebook" /&gt;
                      &lt;Left&gt;&lt;Text style={{ marginLeft: 2 ,fontSize: 12}}&gt;Share it&lt;/Text&gt;&lt;/Left&gt;
                  &lt;/Button&gt;
                &lt;/Body&gt;
              &lt;/CardItem&gt;
           &lt;/Card&gt;
        &lt;/Content&gt;
      &lt;/Container&gt;
    )
  }
}
&nbsp;
export default DetailCollection
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 04 2017 15:09:03 GMT+0700 (WIB)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
