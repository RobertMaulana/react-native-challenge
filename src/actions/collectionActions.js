import { GET_COLLECTIONS } from './constants'
import axios from 'axios'

export const getCollections = collections => ({
  type: GET_COLLECTIONS,
  payload: collections
})

export const fetchCollections = () => (
  dispatch =>
  // axios.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50',
  // {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  // .then(function (response) {
  //   dispatch(getCollections(response.data.collections))
  // })
  // .catch(function (err) {
  //   console.log(err.message);
  // })
  fetch('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50',
  {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
  .then(res => res.json())
  .then(data => dispatch(getCollections(data.collections)))
  .catch(function (err) {
    console.log(err.message);
  })
)
