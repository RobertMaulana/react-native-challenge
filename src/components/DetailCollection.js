import React, {Component} from 'react'
import {
  Image
} from 'react-native'
import moment from 'moment'
import { Container, Content, Card, CardItem, Thumbnail, Text, Button, Icon, Left, Body } from 'native-base';

class DetailCollection extends Component {
  constructor(props){
    super(props)
    this.state = {
    }
  }
  static navigationOptions = {
    title: 'Detail',
  }
  render(){
    const { params } = this.props.navigation.state;
    const restaurant = params.collection;

    return (
      <Container>
        <Content>
            <Card style={{ flex: 0 }}>
              <CardItem>
                <Left>
                  <Thumbnail style={{width: 50, height:50}} source={{uri: 'https://media.glassdoor.com/sql/515676/zomato-squarelogo-1466750863527.png'}}/>
                  <Body>
                      <Text>{restaurant.title}</Text>
                      <Text note>{ moment().format("DD MMM YYYY") }</Text>
                  </Body>
                </Left>
              </CardItem>
              <CardItem>
                <Body>
                  <Image style={{ width: 340, height: 200}} source={{uri: restaurant.image_url}}/>
                  <Text>
                      {restaurant.description}
                  </Text>
                  <Button transparent textStyle={{color: '#87838B'}}>
                      <Icon name="logo-facebook" />
                      <Left><Text style={{ marginLeft: 2 ,fontSize: 12}}>Share it</Text></Left>
                  </Button>
                </Body>
              </CardItem>
           </Card>
        </Content>
      </Container>
    )
  }
}

export default DetailCollection
