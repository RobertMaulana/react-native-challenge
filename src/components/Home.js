import React, {Component} from 'react'
import {
    View,
    Image,
    ScrollView,
    TouchableOpacity
} from 'react-native'
import axios from 'axios'
import { Container, Content, List, ListItem, Thumbnail, Text, Body, Left, Icon, Spinner } from 'native-base';
import { connect } from 'react-redux'

import { fetchCollections } from '../actions'

class Home extends Component {
  static navigationOptions = {
    title: 'Zomato Collections',
  };
  constructor(props){
    super(props)
    this.state = {
      collections: []
    }
  }

  componentDidMount(){
    // let self = this
    // axios.get('https://developers.zomato.com/api/v2.1/collections?city_id=1&count=50', {headers: {"user-key": "ec774bd6f296b71e5ff539dadf4232d8"}})
    // .then(function (response) {
    //   self.setState({restaurants: response.data.collections})
    // })
    // .catch(function (error) {
    //   console.log(error);
    // });
    this.props.fetchCollections()
  }

  render(){
    const { navigate } = this.props.navigation;
    const { headerStyle, textStyle, titleStyle, titleStyleAlignment } = styles
    const { collections } = this.props

    return (
      <Container>
      { collections.length === 0 && <Container>
        <Content>
          <Spinner color='orange' />
        </Content>
      </Container>}
      <Container style={{padding: 5}}>
       <Content>
         <ScrollView>
           <List>
         {collections.map((collection) => {
           return (
            <TouchableOpacity
              key={collection.collection.collection_id}
              onPress={() => navigate('Detail', collection)}
              style={{flexDirection: 'row', marginBottom: 5, borderBottomWidth: 1, paddingBottom: 5, borderBottomColor: '#d0d4db'}}
            >
              <Thumbnail square size={80} source={{uri: collection.collection.image_url}} />
                <Body style={{paddingLeft: 7, alignItems: 'flex-start'}}>
                  <Text>{collection.collection.title}</Text>
                  <Text note style={{fontSize: 12}}>{collection.collection.description}</Text>
              </Body>
            </TouchableOpacity>
         )
         })}
      </List>
         </ScrollView>

       </Content>
     </Container>
     </Container>
    )
  }
}

const styles = {
  headerStyle : {
    paddingTop: '10%',
    alignItems: 'center',
    backgroundColor: "#7b1fa2",
    height: 70,
    shadowColor: '#000',
    shadowOffset: {width: 0, height: 2},
    shadowOpacity: 0.2,
    elevation: 5,
    position: 'relative'
  },
  textStyle: {
    color: "#FFF",
    fontSize: 20
  },
  titleStyleAlignment: {
    width: '100%',
    alignItems: 'center',
    borderBottomColor: '#e2e2e2',
    borderBottomWidth: 1
  }
}

const mapDispatchToProps = dispatch => ({
  fetchCollections: () => dispatch(fetchCollections())
})

const mapStateToProps = state => ({
  collections: state.collections
})

export default connect(mapStateToProps, mapDispatchToProps)(Home)
