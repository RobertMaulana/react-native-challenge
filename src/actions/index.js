import { fetchCollections } from './collectionActions'

export {
  fetchCollections
}
