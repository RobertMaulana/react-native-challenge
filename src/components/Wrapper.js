import React, {Component} from 'react';
import Home from './Home';
import DetailCollection from './DetailCollection'
import {
  StackNavigator
} from 'react-navigation'

const Wrapper = StackNavigator({
  Main: {screen: Home},
  Detail: {screen: DetailCollection}
})
export default Wrapper
