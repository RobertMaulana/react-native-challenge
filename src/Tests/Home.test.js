import React from 'react'
import { shallow } from 'enzyme'
import {
  TouchableOpacity
} from 'react-native'
import { List, Thumbnail, Text, Body } from 'native-base'

import {Home} from '../components/Home'
import { fetchCollections } from '../actions/collectionActions'
import { GET_COLLECTIONS } from '../actions/constants'

describe('<Home />', () => {
  it('should render correctly', () => {
    const home = shallow(
      <Home navigation={{navigate: []}} collections={[]} />
    )
    expect(home).toMatchSnapshot()
  })

  // it("should get function getCollections", () => {
  //   const home = shallow(
  //     <Home navigation={{navigate: []}} collections={[]}/>
  //   )
  //   expect(fetchCollections()).toEqual(
  //       (dispatch) => dispatch(fetchCollections())
  //   )
  //   expect(fetchCollections()).toMatchSnapshot()
  // })

})
const collections = [{}]

const Detail = (props) => {
  return (
    <List>
      {collections.map((collection) => {
        return (
         <TouchableOpacity
           key={props.key}
           onPress={props.onPress}
           style={props.style}
         >
           <Thumbnail square size={80} source={{uri: ''}} />
             <Body style={{paddingLeft: 7, alignItems: 'flex-start'}}>
               <Text>{''}</Text>
               <Text note style={{fontSize: 12}}>{''}</Text>
           </Body>
         </TouchableOpacity>
      )
      })}
   </List>
  )
}

it('pressed', () => {
  const button = shallow(
    <Detail
      key={'key'}
      onPress={() => {}}
      style={{flexDirection: 'row', marginBottom: 5, borderBottomWidth: 1, paddingBottom: 5, borderBottomColor: '#d0d4db'}}
    />
  )
  expect(button).toMatchSnapshot()
})
