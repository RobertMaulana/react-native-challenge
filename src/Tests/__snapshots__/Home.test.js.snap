// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(Container)
    virtual={undefined}
>
    <Styled(Container)
        virtual={undefined}
    >
        <Styled(Content)
            virtual={undefined}
        >
            <Styled(Spinner)
                color="orange"
                virtual={undefined}
            />
        </Styled(Content)>
    </Styled(Container)>
    <Styled(Container)
        style={
            Object {
                "padding": 5,
              }
        }
        virtual={undefined}
    >
        <Styled(Content)
            virtual={undefined}
        >
            <ScrollViewMock>
                <Styled(List)
                    virtual={undefined}
                >
                    
                </Styled(List)>
            </ScrollViewMock>
        </Styled(Content)>
    </Styled(Container)>
</Styled(Container)>,
  "nodes": Array [
    <Styled(Container)
      virtual={undefined}
>
      <Styled(Container)
            virtual={undefined}
      >
            <Styled(Content)
                  virtual={undefined}
            >
                  <Styled(Spinner)
                        color="orange"
                        virtual={undefined}
                  />
            </Styled(Content)>
      </Styled(Container)>
      <Styled(Container)
            style={
                  Object {
                        "padding": 5,
                      }
            }
            virtual={undefined}
      >
            <Styled(Content)
                  virtual={undefined}
            >
                  <ScrollViewMock>
                        <Styled(List)
                              virtual={undefined}
                        >
                              
                        </Styled(List)>
                  </ScrollViewMock>
            </Styled(Content)>
      </Styled(Container)>
</Styled(Container)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Home
        collections={Array []}
        navigation={
                Object {
                        "navigate": Array [],
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Home {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "collections": Array [],
          "navigation": Object {
            "navigate": Array [],
          },
        },
        "refs": Object {},
        "state": Object {
          "collections": Array [],
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(Container)
          virtual={undefined}
>
          <Styled(Container)
                    virtual={undefined}
          >
                    <Styled(Content)
                              virtual={undefined}
                    >
                              <Styled(Spinner)
                                        color="orange"
                                        virtual={undefined}
                              />
                    </Styled(Content)>
          </Styled(Container)>
          <Styled(Container)
                    style={
                              Object {
                                        "padding": 5,
                                      }
                    }
                    virtual={undefined}
          >
                    <Styled(Content)
                              virtual={undefined}
                    >
                              <ScrollViewMock>
                                        <Styled(List)
                                                  virtual={undefined}
                                        >
                                                  
                                        </Styled(List)>
                              </ScrollViewMock>
                    </Styled(Content)>
          </Styled(Container)>
</Styled(Container)>,
        "_debugID": 2,
        "_renderedOutput": <Styled(Container)
          virtual={undefined}
>
          <Styled(Container)
                    virtual={undefined}
          >
                    <Styled(Content)
                              virtual={undefined}
                    >
                              <Styled(Spinner)
                                        color="orange"
                                        virtual={undefined}
                              />
                    </Styled(Content)>
          </Styled(Container)>
          <Styled(Container)
                    style={
                              Object {
                                        "padding": 5,
                                      }
                    }
                    virtual={undefined}
          >
                    <Styled(Content)
                              virtual={undefined}
                    >
                              <ScrollViewMock>
                                        <Styled(List)
                                                  virtual={undefined}
                                        >
                                                  
                                        </Styled(List)>
                              </ScrollViewMock>
                    </Styled(Content)>
          </Styled(Container)>
</Styled(Container)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Home
    collections={Array []}
    navigation={
        Object {
            "navigate": Array [],
          }
    }
/>,
}
`;

exports[`pressed 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Styled(List)
    virtual={undefined}
>
    <TouchableOpacity
        activeOpacity={0.2}
        focusedOpacity={0.7}
        onPress={[Function]}
        style={
            Object {
                "borderBottomColor": "#d0d4db",
                "borderBottomWidth": 1,
                "flexDirection": "row",
                "marginBottom": 5,
                "paddingBottom": 5,
              }
        }
    >
        <Styled(Thumbnail)
            size={80}
            source={
                Object {
                    "uri": "",
                  }
            }
            square={true}
            virtual={undefined}
        />
        <Styled(Body)
            style={
                Object {
                    "alignItems": "flex-start",
                    "paddingLeft": 7,
                  }
            }
            virtual={undefined}
        >
            <Styled(Text)
                virtual={undefined}
            />
            <Styled(Text)
                note={true}
                style={
                    Object {
                        "fontSize": 12,
                      }
                }
                virtual={undefined}
            />
        </Styled(Body)>
    </TouchableOpacity>
</Styled(List)>,
  "nodes": Array [
    <Styled(List)
      virtual={undefined}
>
      <TouchableOpacity
            activeOpacity={0.2}
            focusedOpacity={0.7}
            onPress={[Function]}
            style={
                  Object {
                        "borderBottomColor": "#d0d4db",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "marginBottom": 5,
                        "paddingBottom": 5,
                      }
            }
      >
            <Styled(Thumbnail)
                  size={80}
                  source={
                        Object {
                              "uri": "",
                            }
                  }
                  square={true}
                  virtual={undefined}
            />
            <Styled(Body)
                  style={
                        Object {
                              "alignItems": "flex-start",
                              "paddingLeft": 7,
                            }
                  }
                  virtual={undefined}
            >
                  <Styled(Text)
                        virtual={undefined}
                  />
                  <Styled(Text)
                        note={true}
                        style={
                              Object {
                                    "fontSize": 12,
                                  }
                        }
                        virtual={undefined}
                  />
            </Styled(Body)>
      </TouchableOpacity>
</Styled(List)>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Detail
        onPress={[Function]}
        style={
                Object {
                        "borderBottomColor": "#d0d4db",
                        "borderBottomWidth": 1,
                        "flexDirection": "row",
                        "marginBottom": 5,
                        "paddingBottom": 5,
                      }
        }
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "onPress": [Function],
          "style": Object {
            "borderBottomColor": "#d0d4db",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "marginBottom": 5,
            "paddingBottom": 5,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Styled(List)
          virtual={undefined}
>
          <TouchableOpacity
                    activeOpacity={0.2}
                    focusedOpacity={0.7}
                    onPress={[Function]}
                    style={
                              Object {
                                        "borderBottomColor": "#d0d4db",
                                        "borderBottomWidth": 1,
                                        "flexDirection": "row",
                                        "marginBottom": 5,
                                        "paddingBottom": 5,
                                      }
                    }
          >
                    <Styled(Thumbnail)
                              size={80}
                              source={
                                        Object {
                                                  "uri": "",
                                                }
                              }
                              square={true}
                              virtual={undefined}
                    />
                    <Styled(Body)
                              style={
                                        Object {
                                                  "alignItems": "flex-start",
                                                  "paddingLeft": 7,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        note={true}
                                        style={
                                                  Object {
                                                            "fontSize": 12,
                                                          }
                                        }
                                        virtual={undefined}
                              />
                    </Styled(Body)>
          </TouchableOpacity>
</Styled(List)>,
        "_debugID": 4,
        "_renderedOutput": <Styled(List)
          virtual={undefined}
>
          <TouchableOpacity
                    activeOpacity={0.2}
                    focusedOpacity={0.7}
                    onPress={[Function]}
                    style={
                              Object {
                                        "borderBottomColor": "#d0d4db",
                                        "borderBottomWidth": 1,
                                        "flexDirection": "row",
                                        "marginBottom": 5,
                                        "paddingBottom": 5,
                                      }
                    }
          >
                    <Styled(Thumbnail)
                              size={80}
                              source={
                                        Object {
                                                  "uri": "",
                                                }
                              }
                              square={true}
                              virtual={undefined}
                    />
                    <Styled(Body)
                              style={
                                        Object {
                                                  "alignItems": "flex-start",
                                                  "paddingLeft": 7,
                                                }
                              }
                              virtual={undefined}
                    >
                              <Styled(Text)
                                        virtual={undefined}
                              />
                              <Styled(Text)
                                        note={true}
                                        style={
                                                  Object {
                                                            "fontSize": 12,
                                                          }
                                        }
                                        virtual={undefined}
                              />
                    </Styled(Body)>
          </TouchableOpacity>
</Styled(List)>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Detail
    onPress={[Function]}
    style={
        Object {
            "borderBottomColor": "#d0d4db",
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "marginBottom": 5,
            "paddingBottom": 5,
          }
    }
/>,
}
`;
